inc/class-upgrade.php

```diff
@@ -1575,4 +1588,56 @@ static function ( $duplicate ) use ( $object_type ) {
 		);
 		// phpcs:enable
 	}
+
+	/**
+	 * Removes all indexables for attachments.
+	 *
+	 * @return void
+	 */
+	private function remove_attachment_indexables() {
+		global $wpdb;
+
+		// If migrations haven't been completed successfully the following may give false errors. So suppress them.
+		$show_errors       = $wpdb->show_errors;
+		$wpdb->show_errors = false;
+
+		$indexable_table = Model::get_table_name( 'Indexable' );
+
+		$delete_query = "DELETE FROM $indexable_table WHERE object_sub_type = 'attachment'";
```

Comment:
Just to be sure, I'd also add a `AND object_type = 'post'` here. There could be term indexables for a attachment taxonomy. It also opens up the possibility to use the `object_type_and_sub_type` db index